internal/core/src/common/VectorTrait.h

```diff
@@ -52,7 +53,8 @@ constexpr bool IsVector = std::is_base_of_v<VectorTrait, T>;
 template <typename T>
 constexpr bool IsScalar =
     std::is_fundamental_v<T> || std::is_same_v<T, std::string> ||
-    std::is_same_v<T, Json> || std::is_same_v<T, std::string_view>;
+    std::is_same_v<T, Json> || std::is_same_v<T, std::string_view> ||
+    std::is_same_v<T, Array> || std::is_same_v<T, proto::plan::Array>;
```

Comment:
why add a proto type definition?

Reply:
because we supported the `array/json_array == [1,2,3]`, the element type is proto::plan::Array.